x/payment/keeper/msg_server_withdraw.go

```diff
@@ -4,15 +4,49 @@ import (
 	"context"
 
 	"cosmossdk.io/errors"
+	"cosmossdk.io/math"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
 
 	"github.com/bnb-chain/greenfield/x/payment/types"
 )
 
 func (k msgServer) Withdraw(goCtx context.Context, msg *types.MsgWithdraw) (*types.MsgWithdrawResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
-	// check stream record
+	creator := sdk.MustAccAddressFromHex(msg.Creator)
+
+	params := k.GetParams(ctx)
+	if ctx.IsUpgraded(upgradetypes.Nagqu) {
+		if msg.From == "" { // withdraw from the locked one
+
+			delayedWithdrawal, found := k.GetDelayedWithdrawalRecord(ctx, creator)
```

Comment:
Do we allow the `Amount` to be zero, if it is withdrawn from the locked one? or shall we enforce it to be zero, otherwise it will be a bit confusing.